smarts/core/vehicle.py

```diff
@@ -507,9 +491,9 @@ def step(self, current_simulation_time):
     def control(self, *args, **kwargs):
         self._chassis.control(*args, **kwargs)
 
-    def sync_to_panda3d(self):
+    def sync_to_renderer(self):
         pos, heading = self._chassis.pose.as_panda3d()
-        self._np.setPosHpr(*pos, heading, 0, 0)
+        self._renderer_path.setPosHpr(*pos, heading, 0, 0)
```

Comment:
ok, I think I agree (I was, overall, unhappy with leaving `_renderer_path` in the `Vehicle` class, although moving it would have caused more/bigger ripples. 

 I assume you are suggesting that we add a dictionary to track vehicle nodes to the `Renderer` class, right?  (This can be keyed on the vehicle-id (`vid`) and populated by the call to `create_vehicle()` there, which I'm thinking to rename `create_vehicle_node` btw,)  This would require forwarding the `sync_to_renderer()` call to the `Renderer` object from each `Vehicle` object.  All of this is doable and cleaner, in my opinion.  So assuming this is what you meant, we agree.

I had considered this path, but stopped short when I saw `_follow_vehicle()` in the camera sensors (`sensors.py`) which is uglier / more tangled.  That was enough to scare me away from trying to do the above because I realized to truly get it "clean" would be harder.  Of course I could still do what I wrote in the previous paragraph and leave `follow_camera()` alone, but it seems to undermine the goal . Do you have thoughts/suggestions about this?

Reply:
> I assume you are suggesting that we add a dictionary to track vehicle nodes to the Renderer class, right? (This can be keyed on the vehicle-id (vid) and populated by the call to create_vehicle() there, which I'm thinking to rename create_vehicle_node btw,) This would require forwarding the sync_to_renderer() call to the Renderer object from each Vehicle object. All of this is doable and cleaner, in my opinion. So assuming this is what you meant, we agree.
Yes, renderables should submit to the renderer. I might even go further and remove the Renderer reference from the vehicle and instead force a batched position update.

I think leaving the rest of the camera alone could be fine after removing the `vehicle.renderable_path` use, cameras tend to have privileged access to the renderer. It is usually fine to allow that privileged access because when you manipulate a camera it should generally have no other effect except for a change in the attributes used by the renderer. I might prefer calling the guts of the `_wait_for_ram_image()` inside the renderer as well rather than the camera but it is fairly contained for now.

Reply:
OK.  Will do something along these lines.